ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -674,12 +556,15 @@ public boolean hasFailed() {
       return totalReplicaCount - failedCount < Math.max(totalReplicaCount - 1,
           routerConfig.routerPutSuccessTarget + disabledCount);
     } else {
-      // if there is no possible way to use the remaining replicas to meet the success target,
-      // deem the operation a failure.
-      if (replicaInPoolOrFlightCount + replicaSuccessCount < replicaSuccessTarget) {
+      // If blob is not found in originating DC, we can terminate the operation early. Adding
+      // routerConfig.routerOperationTrackerTerminateOnNotFoundEnabled as a safe guard to search replicas in all colos
+      // before terminating the operation.
+      if (routerConfig.routerOperationTrackerTerminateOnNotFoundEnabled && hasFailedOnNotFound()) {
```

Comment:
Do we plan to enable routerOperationTrackerTerminateOnNotFoundEnabled on delivery?

Reply:
Yeah, was thinking that we could enable it.